src/runtime/threading_backend.cc

```diff
@@ -279,11 +280,7 @@ void Yield() { std::this_thread::yield(); }
 /*!
  * \bief Set the maximum number of available cores.
  */
-void SetMaxConcurrency(int value) {
-  if (value > 0) {
-    max_concurrency = value;
-  }
-}
+void SetMaxConcurrency(int value) { max_concurrency = std::max(0, value); }
```

Comment:
If value is negative should we throw an error instead?

Also behavior is slightly different; In the old code, if value <= 0 it does not change `max_concurrency`.  Here it sets it to 0.

Reply:
> If value is negative should we throw an error instead?

agree, about the error report , for that existing logic just using WARNING level log, to keep logic consistence, the new logic use a LOG(WARNING) to report the error.

>Also behavior is slightly different; In the old code, if value <= 0 it does not change max_concurrency. Here it sets it to 0.

fixed,  when value is negative the new logic will just throw a warning then return.